teamcity: failed tests on master: test/TestLogic, test/TestLogic: TestLogic/default, test/TestLogic: TestLogic/default: TestLogic/default/database, test/TestLogic, test/TestLogic: TestLogic/default, test/TestLogic: TestLogic/default: TestLogic/default/database

The following tests appear to have failed:

[#525846](https://teamcity.cockroachdb.com/viewLog.html?buildId=525846):


```
--- FAIL: test/TestLogic (115.240s)
test_log_scope.go:81: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestLogic836221529
test_log_scope.go:62: use -show-logs to present logs inline
--- FAIL: test/TestLogic: TestLogic/default (0.310s)
None
--- FAIL: test/TestLogic: TestLogic/default: TestLogic/default/database (0.810s)
logic.go:2115: 
	testdata/logic_test/database:164: 
	expected success, but found
	pq: user testuser does not have DROP privilege on database privs
logic.go:2131: 
	testdata/logic_test/database:164: error while processing
logic.go:2132: 
	testdata/logic_test/database:164: error in statement, skipping to next file
--- FAIL: test/TestLogic (115.240s)
test_log_scope.go:81: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestLogic836221529
test_log_scope.go:62: use -show-logs to present logs inline
--- FAIL: test/TestLogic: TestLogic/default (0.310s)
None
--- FAIL: test/TestLogic: TestLogic/default: TestLogic/default/database (0.810s)
logic.go:2115: 
	testdata/logic_test/database:164: 
	expected success, but found
	pq: user testuser does not have DROP privilege on database privs
logic.go:2131: 
	testdata/logic_test/database:164: error while processing
logic.go:2132: 
	testdata/logic_test/database:164: error in statement, skipping to next file

```

Please assign, take a look and update the issue accordingly.

--- a/pkg/sql/logictest/testdata/logic_test/database
+++ b/pkg/sql/logictest/testdata/logic_test/database
@@ -161,5 +161,11 @@ GRANT DROP ON DATABASE privs TO testuser
 
 user testuser
 
+# Wait for gossip to propagate the new version of the database descriptor.
+#
+# TODO: block the `GRANT` statement until the new version has been propagated.
+# See #22841.
+sleep 250ms
+
 statement ok
 DROP DATABASE privs CASCADE

